src/frontend/screens/ObservationFields/Number.tsx

```diff
@@ -20,7 +20,8 @@ export const Number = React.memo<{field: Field}>(({field}) => {
             field.tagKey,
             newVal
               .replace(/[^0-9.-]/g, '') // Allow digits, decimal, and negative sign
-              .replace(/(?!^)-/g, ''), // Remove any minus sign that is not at the start
+              .replace(/(?!^)-/g, '') // Remove any minus sign that is not at the start
```

Comment:
This works, but you might consider `.replace(/^(-?)(\d*)(\.?\d*).*/, '$1$2$3') `instead to have one line for the regex. It does the same thing but in one line instead of two...